### DISCUSSION TITLE
is using one server deployment safe to do?

### DISCUSSION THREAD

**Discussion by laptopmutia** [basecamp/kamal#317] [Q&A]
all the tutorial example is showing me that to use multiple servers and managed database
and dhh seems also against 1 server setup as per discussion in here https://github.com/mrsked/mrsk/issues/3 and recommend to use sqlite

but that discussions is prior to this PR https://github.com/mrsked/mrsk/pull/23

but there is a tutorial that I found that tell me how to do deployment with mrsk on one server https://www.kartikey.dev/2023/04/05/how-to-deploy-rails-app-and-postgres-with-mrsk-on-single-server.html

my questions is is this safe to do on production?

currently I want to deploy a not that big app but it not fit with [sqlite use case guide](https://www.sqlite.org/whentouse.html) and the company that hire me only provide me with one VM machine to deploy to

**Comment by MikeBinstock** [comment]
works perfect with one server

**Reply by curtisspendlove** [reply]
So far, so good. You can deploy your accessories to the same server as the web. Just keep in mind you have all the downsides of a single point of failure in that mode. Ensure you have frequent backups and monitor metrics (cpu, memory, storage). 

If you see the application or database bottlenecking you’ll want to take appropriate measures. But you can scale vertically quite a bit by throwing more CPU and RAM at a single VM. Scaling horizontally too soon can be premature optimization. 

Not every application needs shared database servers fronted by a dozen load-balanced application servers.  

I’m running a small application on a $7/month digital ocean droplet. It just depends on your needs and what you have available.
